### ISSUE TITLE
JS devDependencies outdated

### ISSUE THREAD

**Issue by cschwartz** [gophish/gophish#1428]
**Thanks for reaching out! We're happy to help resolve issues as quickly as possible.**

**Please use this template when creating a new issue. If you do not follow this template format, your issue may be closed without comment.**

What version of Gophish are you using?: Master branch

Brief description of the issue: 
First of all, thanks for all the continued great work with GoPhish.

I was just setting up my environment to look if I could contribute, when I realized that the devDependencies used to build GoPhishs frontend are outdated and include deprecated packages (see below for the npm output).

Unless there is a reason why you want to keep the current version of packages around, I could provide a pull request to update all used packages to their current version (note, that the current version of Gulp changed the syntax to define tasks, so this would probably include some changes in the `gulpfile.js`).

What are you expecting to see happen? :
No deprecated packages are being used.

What are you seeing happen?
Deprecated packages are being used.

Please provide any terminal output that may be relevant below:

```
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated wreck@6.3.0: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated boom@2.10.1: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated hoek@2.16.3: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated natives@1.1.6: This module relies on Node.js's internals and will break at some point. Do not use it, and update to graceful-fs@4.x.
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm notice created a lockfile as package-lock.json. You should commit this file.
added 430 packages from 232 contributors and audited 2686 packages in 5.679s
found 12 vulnerabilities (4 low, 3 moderate, 5 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

Please provide as many steps as you can to reproduce the problem:
 * `go get github.com/gophish/gophish`
 * `npm install`

**Comment by jordan-wright** [comment]
Hi @cschwartz,

This admittedly doesn't surprise me. I'd be happy to accept a pull request that updates the dependency versions as well as the needed Gulp task fixes.

That said, if you do want to commit a pull request, I'd ask that you please don't commit the new minified JS files. Rather, please just commit the new `package.json`, `yarn.lock`, and `Gulpfile` (basically just the metadata). I'll merge those into a branch and compile the JS on my local machine.

Thanks for the heads up on this!

**Comment by jordan-wright** [comment]
Closed via #1440
